homeassistant/components/weather/demo.py

```diff
@@ -0,0 +1,70 @@
+"""
+Demo platform that offers fake meteorological data.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/demo/
+"""
+from homeassistant.components.weather import WeatherCondition
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Demo weather."""
+    add_devices([DemoWeather('sunshine', 21, 78, 1099, 3)])
+
+
+# pylint: disable=too-many-arguments
+class DemoWeather(WeatherCondition):
+    """Representation of a weather condition."""
+
+    def __init__(self, condition, temperature, humidity, pressure, wind_speed):
+        """Initialize the Demo weather."""
+        self._condition = condition
+        self._temperature = temperature
+        self._humidity = humidity
+        self._pressure = pressure
+        self._wind_speed = wind_speed
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return 'Demo weather'
+
+    @property
+    def should_poll(self):
+        """No polling needed for a demo weather condition."""
+        return False
+
+    @property
+    def temperature(self):
+        """Return the temperature."""
+        return self._temperature
+
+    @property
+    def humidity(self):
+        """Return the humidity."""
+        return self._humidity
+
+    @property
+    def wind_speed(self):
+        """Return the wind speed."""
+        return self._wind_speed
+
+    @property
+    def pressure(self):
+        """Return the wind speed."""
+        return self._pressure
+
+    @property
+    def state(self):
```

Comment:
I think that it's save to say that temperature and humidity are available by the providers we using now and will in the future.

There is a mapping solution present now that should make sure that we limit the conditions from the platforms to approx. a dozen "Home Assistant conditions".